feature/security/gnsi/credentialz/tests/password_console_login/password_console_login_test.go

```diff
@@ -0,0 +1,310 @@
+		}
+	default:
+		t.Fatalf("Unsupported vendor %s for deviation 'deviation_native_users'", dut.Vendor())
+	}
+}
+
+func setupUser(t *testing.T, dut *ondatra.DUTDevice) {
+	auth := &oc.System_Aaa_Authentication{}
+	auth.GetOrCreateUser(username)
+
+	gnmi.Update(t, dut, gnmi.OC().System().Aaa().Authentication().Config(), auth)
+
+	if deviations.SetNativeUser(dut) {
+		// probably all vendors need to handle this since the user should have a role attached to
+		// it allowing us to login via ssh/console/whatever
+		createNativeRole(t, dut, "credz-fp-test")
+	}
+}
+
+func rotateUserCredential(t *testing.T, dut *ondatra.DUTDevice) {
+	credzClient := dut.RawAPIs().GNSI(t).Credentialz()
+
+	credzRotateClient, err := credzClient.RotateAccountCredentials(context.Background())
+	if err != nil {
+		t.Fatalf("failed fetching credentialz rotate account credentials client, error: %s", err)
+	}
+
+	req := &credentialz.RotateAccountCredentialsRequest{
+		Request: &credentialz.RotateAccountCredentialsRequest_Password{
+			Password: &credentialz.PasswordRequest{
+				Accounts: []*credentialz.PasswordRequest_Account{
+					{
+						Account: username,
+						Password: &credentialz.PasswordRequest_Password{
+							Value: &credentialz.PasswordRequest_Password_Plaintext{
+								Plaintext: password,
+							},
+						},
+						Version:   passwordVersion,
+						CreatedOn: passwordCreatedOn,
+					},
+				},
+			},
+		},
+	}
+
+	err = credzRotateClient.Send(req)
+	if err != nil {
+		t.Fatalf("failed sending credentialz rotate account credentials request, error: %s", err)
+	}
+
+	resp, err := credzRotateClient.Recv()
+	if err != nil {
+		t.Fatalf("failed receiving credentialz rotate request, error: %s", err)
+	}
+
+	t.Logf("got credentialz rotate request response: %s", resp)
+
+	err = credzRotateClient.Send(&credentialz.RotateAccountCredentialsRequest{
+		Request: &credentialz.RotateAccountCredentialsRequest_Finalize{
+			Finalize: req.GetFinalize(),
+		},
+	})
+	if err != nil {
+		t.Fatalf("failed sending credentialz rotate account credentials finalize request, error: %s", err)
+	}
+}
+
+func assertTelemetry(t *testing.T, dut *ondatra.DUTDevice) {
+	userState := gnmi.Get(t, dut, gnmi.OC().System().Aaa().Authentication().User(username).State())
+
+	actualPasswordVersion := userState.GetPasswordVersion()
+
+	if actualPasswordVersion != passwordVersion {
+		t.Fatalf(
+			"telemetry reports password version is not correctn\tgot: %s\n\twant: %s",
+			actualPasswordVersion, passwordVersion,
+		)
+	}
+
+	actualPasswordCreatedOn := userState.GetPasswordCreatedOn()
+
+	if time.Unix(int64(passwordCreatedOn), 0).Compare(time.Unix(0, int64(actualPasswordCreatedOn))) != 0 {
+		t.Fatalf(
+			"telemetry reports password created on is not correct\n\tgot: %d\n\twant: %d",
+			actualPasswordCreatedOn, passwordCreatedOn,
+		)
+	}
+}
+
+func tryLogin(t *testing.T, addr, loginUser, loginPassword string, expectFail bool) {
+	client, err := ssh.Dial(
```

Comment:
Maybe I'm confused, but to me "console" and "ssh" are distinct things and there should be no ssh dial for a console test.

Reply:
Agree this should be a console dial & not ssh. Currently featureprofiles does not have an ondatra binding implementation for a console client & as per the discussion on this [thread](https://github.com/openconfig/featureprofiles/issues/2637), it seems there is no proper way to retrieve console/ssh endpoints from the binding. Could you please suggest how we can go about with this test? Here is a [buganizer](https://issuetracker.google.com/304734163) opened for this requirement.

Reply:
Cool, thanks for that link. I commented over on that bug and can follow up with the assignee. That's a blocker for console tests in my opinion.